### PR TITLE
Fix warnings from cpp-check 1.73

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This fixes some of the warnings found with cpp-check 1.73. 

**To test:**

For most of these changes, code review should be sufficient. Also see the comments below about a "fix" that breaks several unit tests and using a smart pointer to eliminate an if statement cppcheck was warning about.

This is a small change with no issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/FindPeaks.cpp [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ int FindPeaks::findPeakBackground(const MatrixWorkspace_sptr &input,
         "No 7th column for use FindPeakBackground result or not. ");
 
   if (peaklisttablews->rowCount() > 0) {
-    int fitresult = peaklisttablews->Int(0, 6);
+    fitresult = peaklisttablews->Int(0, 6);
```
Comment (quantumsteve) [inline Framework/Algorithms/src/FindPeaks.cpp]: @peterfpeterson Would you please check if there is an issue setting `fitresult` and the above change is correct. Should this change be in the release notes?

File: MantidPlot/src/ApplicationWindow.cpp [hunk 2]
```diff
@@ -16554,18 +16551,14 @@ void ApplicationWindow::CatalogLogin() {
 }
 
 void ApplicationWindow::CatalogSearch() {
-  if (catalogSearch == NULL || catalogSearch) {
```
Comment (quantumsteve) [inline MantidPlot/src/ApplicationWindow.cpp]: @ElliotAOram
Does switching to a smart pointer eliminate the check you added?

Comment (ElliotAOram) [inline MantidPlot/src/ApplicationWindow.cpp]: @quantumsteve 
I don't think I actually wrote this code. I think it's likely that my name is there in the git blame due to a clang format. Regardless, given the nature of the check I think a smart pointer should work.

File: Framework/Algorithms/src/FindPeaks.cpp (hunk 1 above)
Comment (FedeMPouzols) [inline Framework/Algorithms/src/FindPeaks.cpp]: That apparently innocent line is a can of worms: https://github.com/mantidproject/mantid/pull/13667, https://github.com/mantidproject/mantid/issues/13950 (from a coverity issue that caught this).

Comment (peterfpeterson) [inline Framework/Algorithms/src/FindPeaks.cpp]: @FedeMPouzols is right. Unshadowing that variable is bad. I haven't had the time to track down why it is so important, but the fits are significantly better with it there.

Comment (quantumsteve) [inline Framework/Algorithms/src/FindPeaks.cpp]: @FedeMPouzols @peterfpeterson Thanks. I reverted the change and added a comment referencing the issue number above.

File: MantidPlot/src/ApplicationWindow.cpp (hunk 2 above)
Comment (quantumsteve) [inline MantidPlot/src/ApplicationWindow.cpp]: @ElliotAOram Thank you for checking.

**Comment by martyngigg** [comment]
I agree with the all of the changes that have been made. I too have come across the bug in `FindPeaks` that breaks many tests. Should someone have a maintenance issue to investigate this?

**Comment by martyngigg** [comment]
Other than that, let's :shipit:

**Comment by FedeMPouzols** [comment]
We have this issue: #13950, which was tagged as "coverity". I've added "cppcheck". Maybe it can be tackled for the next maintenance period. At the moment we also have the GSL2 changes ongoing. I'd wonder how those changes would interplay with this issue.

**Pull request merged**